cocotb/__init__.py

```diff
@@ -187,6 +187,14 @@ def _reopen_stream_with_buffering(stream_name):
 
 def fork(coro: Union[RunningTask, Coroutine]) -> RunningTask:
     """Schedule a coroutine to be run concurrently. See :ref:`coroutines` for details on its use."""
+    warnings.warn(
+        "cocotb.fork has been deprecated in favor of cocotb.start_soon and cocotb.start.\n"
+        "In most cases you can simply substitute cocotb.fork with cocotb.start_soon.\n"
+        "For more information about when you would want to use cocotb.start see the docs.\n"
+        "https://docs.cocotb.org/en/latest/library_reference.html#cocotb.start",
```

Comment:
https://docs.cocotb.org/en/latest/coroutines.html#concurrent-execution might be better?